File: BlackSwanTest/app/src/main/java/com/tp/projects/blackswantest/MovieTilesAdapter.java
package com.tp.projects.blackswantest;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Tokaji Peter on 21/07/16.
 */
public class MovieTilesAdapter extends RecyclerView.Adapter<MovieTilesAdapter.Viewholder> {

    public static class Viewholder extends RecyclerView.ViewHolder {

        View tile;

        public Viewholder(View itemView) {
            super(itemView);
            tile = itemView;
        }
    }

    Context ctx;
    List<MovieTile> movieList;

    public MovieTilesAdapter(Context context, List<MovieTile> movieTiles) {
        ctx = context;
        movieList = new ArrayList<>();
        movieList.addAll(movieTiles);
    }


    @Override
    public Viewholder onCreateViewHolder(ViewGroup parent, int viewType) {
        View tile = LayoutInflater.from(ctx).inflate(R.layout.movie_tile, parent, false);

        return new Viewholder(tile);
    }

    @Override
    public void onBindViewHolder(Viewholder holder, final int position) {
        ((TextView) holder.tile.findViewById(R.id.movie_title)).setText(movieList.get(position).getTitle());
        ((TextView) holder.tile.findViewById(R.id.movie_popularity)).setText(String.valueOf(movieList.get(position).getPopularity()));
        ((TextView) holder.tile.findViewById(R.id.movie_description)).setText(movieList.get(position).getOverview());
        holder.tile.findViewById(R.id.more_info).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(ctx, MovieDetailsActivity.class);
                intent.putExtra("movie",movieList.get(position));
                ctx.startActivity(intent);
            }
        });

        Picasso.with(ctx)
                .load(movieList.get(position).getPosterPath())
                .placeholder(R.mipmap.ic_launcher)
                .into((ImageView) holder.tile.findViewById(R.id.movie_image));

    }

    @Override
    public int getItemCount() {
        return movieList.size();
    }
}

File: MovieDB/app/src/main/java/com/tp/projects/moviedb/movies/MovieTilesAdapter.kt
package com.tp.projects.moviedb.movies

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso
import com.tp.projects.moviedb.R

/**
 * Created by Tokaji Peter on 21/07/16.
 */
internal class MovieTilesAdapter(private val ctx: Context, private val movieList: List<MovieData>) :
        RecyclerView.Adapter<MovieTilesAdapter.ViewHolder>() {

    internal class ViewHolder(var tile: View) : RecyclerView.ViewHolder(tile)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
            ViewHolder(LayoutInflater.from(ctx).inflate(R.layout.tile_layout, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tile.apply {
            val actualMovie = movieList[position]
            findViewById<TextView>(R.id.tile_title).text = actualMovie.title
            findViewById<TextView>(R.id.tile_popularity).text = "Rating: ${actualMovie.voteAverage}"
            findViewById<TextView>(R.id.tile_description).text = actualMovie.overview
            findViewById<Button>(R.id.more_info).setOnClickListener {
                ctx.startActivity(Intent(ctx, MovieDetailsActivity::class.java).apply {
                    putExtra("movie", actualMovie)
                })
            }
            Picasso.get()
                    .load(movieList[position].posterPath)
                    .placeholder(R.mipmap.ic_launcher)
                    .into(findViewById<ImageView>(R.id.tile_image))
        }
    }

    override fun getItemCount(): Int {
        return movieList.size
    }
}
